extensions/vscode/src/types/messages/hostToWebviewMessages.ts

```diff
@@ -71,15 +72,15 @@ export type RefreshContentRecordDataMsg = AnyHostToWebviewMessage<
   HostToWebviewMessageType.REFRESH_CONTENTRECORD_DATA,
   {
     contentRecords: (ContentRecord | PreContentRecord)[];
-    selectedContentRecordName?: string | null;
+    deploymentSelected: DeploymentSelector | null;
```

Comment:
switch to a DeploymentSelector object, which allows us to select the exact combination of deployment and config. It is unique across all locations within a set of directories, since it contains the actual deployment path on disk.